pkg/operator/operator.go

```diff
@@ -130,10 +133,19 @@ func NewOperator(ctx context.Context, operator *operator.Operator) (context.Cont
 		log.FromContext(ctx).WithValues("kube-dns-ip", kubeDNSIP).V(1).Info("discovered kube dns")
 	}
 
+	//v2
+	//Once everything is migrated we will need to update prometheus metrics to v2
+	cfg := lo.Must(configV2.LoadDefaultConfig(ctx, configV2.WithRetryMaxAttempts(3)))
+	if cfg.Region == "" {
+		log.FromContext(ctx).V(1).Info("retrieving region from IMDS")
+		metaDataClient := imds.NewFromConfig(cfg)
+		cfg.Region = lo.Must(metaDataClient.GetRegion(ctx, nil)).Region
```

Comment:
As a follow-up, we should add in an error message here https://github.com/aws/karpenter-provider-aws/pull/7097/files#diff-99c0290799daf9abc6240df64063e20bfaf67b371577b67ac7eec6f4725622ffR111